fastly/data_source_fastly_domains_v1.go

```diff
@@ -0,0 +1,133 @@
+package fastly
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/fastly/go-fastly/v12/fastly/domainmanagement/v1/domains"
+
+	"github.com/fastly/terraform-provider-fastly/fastly/hashcode"
+)
+
+func dataSourceFastlyDomainsV1() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceFastlyDomainsV1sRead,
+		Schema: map[string]*schema.Schema{
+			"domains": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A domain represents the domain name through which visitors will retrieve content. There can be multiple domains for a service.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fqdn": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The fully-qualified domain name for your domain. Can be created, but not updated.",
```

Comment:
```suggestion
							Description: "The fully-qualified domain name for your domain.",
```

This is a data source, so all schema elements are read-only.

Reply:
Addressed in [9ffc47c](https://github.com/fastly/terraform-provider-fastly/pull/1112/commits/9ffc47ce7c1cbbaf82be398b551fbc17e8e42b8c)